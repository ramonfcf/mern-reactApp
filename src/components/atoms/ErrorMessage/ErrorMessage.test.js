import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import ErrorMessage from "./ErrorMessage";

describe("ErrorMessage component", () => {
  it("matches snapshot", () => {
    const { container } = render(<ErrorMessage error="Test error" />);
    expect(container).toMatchSnapshot();
  });

  it("renders with correct error message", () => {
    render(<ErrorMessage error="Test error" />);
    const errorMessageElement = screen.getByRole("alert");

    expect(errorMessageElement).toHaveTextContent("Error: Test error");
  });
});
