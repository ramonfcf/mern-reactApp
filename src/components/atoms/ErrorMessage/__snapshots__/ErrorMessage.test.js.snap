// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component matches snapshot 1`] = `
<div>
  <div
    class="alert alert-danger"
    role="alert"
  >
    Error: 
    Test error
  </div>
</div>
`;
